src/Bicep.RegistryModuleTool/Proxies/BicepCliProxy.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Exceptions;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+
+namespace Bicep.RegistryModuleTool.Proxies
+{
+    internal sealed class BicepCliProxy
+    {
+        private readonly static Regex BicepBuildWarningRegex = new(
+            @"^([^\s].*)\((\d+)(?:,\d+|,\d+,\d+)?\)\s+:\s+(Warning)\s+([a-zA-Z-\d]+):\s*(.*?)\s+\[(.*?)\]$",
+            RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        private readonly static string[] LineSeperators = new[] { "\r", "\n", "\r\n" };
+
+        private readonly IEnvironmentProxy environmentProxy;
+
+        private readonly IProcessProxy processProxy;
+
+        private readonly IFileSystem fileSystem;
+
+        private readonly ILogger logger;
+
+        public BicepCliProxy(IEnvironmentProxy environmentProxy, IProcessProxy processProxy, IFileSystem fileSystem, ILogger logger)
+        {
+            this.environmentProxy = environmentProxy;
+            this.processProxy = processProxy;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public void Build(string bicepFilePath, string outputFilePath)
+        {
+            this.logger.LogDebug("Building \"{BicepFilePath}\"...", bicepFilePath);
+
+            var bicepCliPath = this.LocateBicepCli();
+            var command = $"build \"{bicepFilePath}\" --outfile \"{outputFilePath}\"";
```

Comment:
>command [](http://example.com/codeflow?start=16&length=7)

There's a CommandBuilder class that can help with escaping btw. I think I'm using it in the repo somewhere.

Reply:
That would be great, but I could not find it. Did you mean the Bicep repo?

Reply:
Sorry I lied. It's called `CommandLineBuilder`. We are using it here: [https://github.com/Azure/bicep/blob/bdda2d0bc6742ba3792a4628f042713c0d746b8b/src/Bicep.MSBuild/Bicep.cs#L28](https://github.com/Azure/bicep/blob/bdda2d0bc6742ba3792a4628f042713c0d746b8b/src/Bicep.MSBuild/Bicep.cs#L28)

It does require taking a dependency on an msbuild NuGet. I'm ok either way.